libs/agno/agno/agent/agent.py

```diff
@@ -4281,18 +4305,24 @@ async def _ahandle_agent_run_paused_stream(
 
         log_debug(f"Agent Run Paused: {run_response.run_id}", center=True, symbol="*")
 
-    def _convert_response_to_structured_format(self, run_response: Union[RunOutput, ModelResponse]):
+    def _convert_response_to_structured_format(self, run_response: Union[RunOutput, ModelResponse], run_context: Optional[RunContext] = None):
+        # run_context override
+        if run_context is not None and run_context.output_schema is not None:
+            output_schema = run_context.output_schema  # type: ignore[assignment]
+        else:
+            output_schema = self.output_schema  # type: ignore[assignment]
```

Comment:
No rather sort this out right at the beginning, so that you can only refer to the run context.

If you can help it, these helper functions should not refer to `self` (general rule) -> Then we can move them out to utils, which is the big win